.github/workflows/dist.yml

```diff
@@ -0,0 +1,30 @@
+name: dist-check
```

Comment:
What worries me about the dist check is that we don't run auto-sklearn afterwards. This is mostly to check that we don't forget to include anything that is vital to run in the distribution. Can you maybe factor that into the actual tests (similar to  codecov)